docs/developer/architecture/development-plugin-saved-objects.asciidoc

```diff
@@ -0,0 +1,205 @@
+[[development-plugin-saved-objects]]
+== Using Saved Objects
+
+Saved Objects allow {kib} plugins to use {es} like a primary
+database. Think of it as an Object Document Mapper for {es}. Once a
+plugin has registered one or more Saved Object types, the Saved Objects client
+can be used to query or perform create, read, update and delete operations on
+each type.
```

Comment:
Not sure if relevant, is it worth to specify that types are not isolated per plugin, meaning that the client can be used to query types registered from other plugins (even in if most cases, this is probably a bad practice)

Reply:
I don't think developers would assume they can rely on such isolation for security and the HTTP API would make it quite obvious that the only protection comes from spaces+security plugin.